### ISSUE TITLE
SetEntityString function not complete?

### ISSUE THREAD

**Issue by AmenMoses** [TheGameCreators/GameGuruRepo#387]
The SetEntityString Lua function sets an entities sound slot string but doesn't actually change the sound file that is loaded.  Is there a second command that needs to be issued after setting the strings to refresh the sounds themselves or is has this part just not been completed yet?

As an experiment I changed the c++ function to load the specified sound slot using the string supplied and it appeared to work ok but I'm not quite sure if this is all that needs doing, i.e. should it also do the steam update bit for multiplayer?  Should it check if the sound is playing and stop it before replacing it?  What about 'cloned' sounds, is it safe to simply delete the old sound or only if it is not 'cloned' or is that already handled lower down?

**Comment by LeeBamberTGC** [comment]
Yes, the sounds are loaded before the main game starts (for performance reasons).  You would need a new LUA command to cause an entity to reload its sounds (but expect an FPS stutter as the file system is hit), and your other concerns over hot swapping a sound are valid. I have marked this as a feature request as its unusual to load anything in the middle of a game running, always better to load at the start of each level or during a transition pause.

**Comment by AmenMoses** [comment]
I'm confused, what is the command for then?  I couldn't find any uses in the standard set of scripts so is it just something you started working on and didn't finish?

(btw, as long as the sound files are small you cannot really notice them being switched but for the piano script the sound loading is done right at the start as part of the initialisation, along with positioning the keys and lid, not during any game play)

If you're happy for me to do so I'll push the changes I've made as is and you can check if you think anything else is required to make the changes 'safe'.

**Comment by LeeBamberTGC** [comment]
The 'SetEntityString' command re-uses the string variables normally reserved for sound filenames for other per-entity purposes. It's a little redundant given that you can simply create your own global string array in LUA and use the entity ID as a subscript to that array.  I suppose if the string being set had a ".WAV" or ".OGG" extension there would be no harm in refreshing the sound file, and we can document it in the global.lua file :)
